src/aux-server/aux-backend/shared/ServerBuilder.ts

```diff
@@ -1264,6 +1275,73 @@ export class ServerBuilder implements SubscriptionLike {
         return this;
     }
 
+    /**
+     * Configures the server to use tigerbeetle for financial transactions.
+     * @param options The options to use.
+     */
+    useTigerBeetle(
+        options: Pick<ServerConfig, 'tigerBeetle'> = this._options
+    ): this {
+        if (!options.tigerBeetle || !options.tigerBeetle._enabled) {
+            console.warn(
+                '[ServerBuilder] TigerBeetle is explicitly disabled or lacks necessary config.'
+            );
+            return this;
+        }
+
+        console.log('[ServerBuilder] Using TigerBeetle Financial Interface.');
+
+        /**
+         * !!! TigerBeetle does not provide a way to check the status of server connections:
+         * * Because of this, misconfiguration can lead to an everlasting retry loop, which can be problematic and use resources unnecessarily.
+         * * To mitigate this, a timeout is set to race a method invocation which proves an established connection during server build.
+         * * If the connection is not proven established within the duration the timeout permits, the server will throw an error (and exit).
+         * * Disconnects (or any network errors) after the server is built will not be detected / handled by this mechanism;
+         *   it serves only to deter building the server with invalid connection parameters.
+         */
+        this._actions.push({
+            priority: 0,
+            action: async () => {
+                const client = createClient({
+                    cluster_id: options.tigerBeetle.clusterId,
+                    replica_addresses: options.tigerBeetle.replicaAddresses,
+                });
+                console.log(
+                    '[ServerBuilder] Connecting to tigerbeetle server.'
+                );
+                try {
+                    await Promise.race([
+                        client.lookupAccounts([0n]),
+                        new Promise((_, rej) => {
+                            setTimeout(() => {
+                                rej(
+                                    new Error(
+                                        'Failed to connect to tigerbeetle server in time.'
+                                    )
+                                );
+                            }, 3000);
+                        }),
+                    ]);
```

Comment:
@LandonSiler We don't need to validate the tiger beetle connection on start. If we are having issues, then the server should start but each request should fail. This helps the system recover from temporary failures and not run into issues where we have to manually restart the server executable.